pkg/adaptor/cloud/ibmcloud/provider.go

```diff
@@ -414,3 +414,11 @@ func (p *ibmcloudVPCProvider) DeleteInstance(ctx context.Context, instanceID str
 func (p *ibmcloudVPCProvider) Teardown() error {
 	return nil
 }
+
+func (p *ibmcloudVPCProvider) Check() error {
+	images := p.serviceConfig.Images.String()
```

Comment:
Possible to check the API-KEY and image existing via image ID?

Reply:
it already be checked at [updateImageList](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/pkg/adaptor/cloud/ibmcloud/provider.go#L125C20-L125C35) inside `NewProvider` function.

Reply:
oh, I see, sounds we don't need extra checking any more because `NewProvider` guarantee the parameters works well? like:
https://github.com/confidential-containers/cloud-api-adaptor/blob/main/cmd/cloud-api-adaptor/main.go#L30
https://github.com/confidential-containers/cloud-api-adaptor/blob/main/pkg/adaptor/cloud/ibmcloud/manager.go#L43
https://github.com/confidential-containers/cloud-api-adaptor/blob/main/pkg/adaptor/cloud/ibmcloud/provider.go#L125
If there is any wrong in `NewProvider`, the provider can not be initialized and the service can not start, therefore the UDS will not be opened. The `uds` has been checked in startup probe.

Reply:
keep the simple checker here to keep consistent with other providers

Reply:
Maybe we don't need a new Check interface but add the check in existing `NewProvider`? It's already there and did some checking.

Reply:
It's reasonable for me to keep a separate check function also if we considering `NewProvider` just set configuratins and `Check` to verify the configurations like do some async http calls.